plugins/hypervisors/kvm/src/com/cloud/hypervisor/kvm/resource/wrapper/LibvirtStartCommandWrapper.java

```diff
@@ -125,9 +125,14 @@ public Answer execute(final StartCommand command, final LibvirtComputingResource
                 // try to patch and SSH into the systemvm for up to 5 minutes
                 for (int count = 0; count < 10; count++) {
                     // wait and try passCmdLine for 30 seconds at most for CLOUDSTACK-2823
-                    libvirtComputingResource.passCmdLine(vmName, vmSpec.getBootArgs());
-                    // check router is up?
-                    final VirtualRoutingResource virtRouterResource = libvirtComputingResource.getVirtRouterResource();
+                    if (libvirtComputingResource.passCmdLine(vmName, vmSpec.getBootArgs())) {
+                        break;
+                    }
```

Comment:
@rhtyd 
we pass cmdline to systemvm multiple times here, is because the operation passCmdLine will always succeed  but actually systemvm cannot get the data via /dev/vport01p1 inside it, so we pass the cmdline to systemvm around every 15 seconds (virtRouterResource.connect) until control ip is reachable.

Reply:
@ustcweizhou the blind waiting and passing cmdline is not necessary now, in the new patch.sh script it waits until it's able to communicate with the qemu-guest-agent running inside the systemvm appliance. See: https://github.com/apache/cloudstack/blob/master/scripts/vm/hypervisor/kvm/patch.sh#L61

Reply:
@ustcweizhou I can reduce that to 3 maybe.
I could consistently reproduce the issue with latest Ubuntu 19.10 with latest Centos7.6 used as kvm host.